lib/ecto/migration/runner.ex

```diff
@@ -275,18 +275,17 @@ defmodule Ecto.Migration.Runner do
         module.after_begin()
       end
 
-      result = apply(module, operation, [])
+      apply(module, operation, [])
+      flush()
 
       if function_exported?(module, :before_commit, 0) do
         module.before_commit()
       end
-
-      result
+      flush() # flush it again if `before_commit/0` queued anything up
```

Comment:
We should move it to inside the function_exported? conditional. WDYT?

Reply:
Agree!

Reply:
Amended